codex-rs/mcp-server/src/lib.rs

```diff
@@ -77,10 +82,26 @@ pub async fn run_main(codex_linux_sandbox_exe: Option<PathBuf>) -> IoResult<()>
         }
     });
 
+    // Parse CLI overrides once so we can reuse them for operations
+    // like `get_auth_status` that need access to the effective config.
+    let cli_kv_overrides = match cli_config_overrides.parse_overrides() {
+        Ok(v) => v,
+        Err(e) => {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                format!("error parsing -c overrides: {e}"),
+            ));
+        }
+    };
+
```

Comment:
FYI, you could do `let cli_kv_overrides = cli_config_overrides.parse_overrides().map_err(...)?;` if you wanted to avoid the `match` and try to be a bit more succinct.